src/vehicle.cpp

```diff
@@ -5109,18 +5110,18 @@ int vehicle::traverse_vehicle_graph( Vehicle *start_veh, int amount, Func action
         return amount;
     }
     // Breadth-first search! Initialize the queue with a pointer to ourselves and go!
-    std::queue< std::pair<Vehicle *, int> > connected_vehs;
-    std::set<Vehicle *> visited_vehs;
-    std::set<tripoint> visted_targets;
-    connected_vehs.push( std::make_pair( start_veh, 0 ) );
+    std::vector< std::pair<Vehicle *, int> > connected_vehs;
+    std::vector<Vehicle *> visited_vehs;
+    std::vector<tripoint> visited_targets;
+    connected_vehs.push_back( std::make_pair( start_veh, 0 ) );
 
     while( amount > 0 && !connected_vehs.empty() ) {
-        auto current_node = connected_vehs.front();
+        auto current_node = connected_vehs.back();
         Vehicle *current_veh = current_node.first;
         int current_loss = current_node.second;
 
-        visited_vehs.insert( current_veh );
-        connected_vehs.pop();
+        visited_vehs.push_back( current_veh );
+        connected_vehs.pop_back();
```

Comment:
I want to confirm my understanding here that doing this is okay:

```
auto someElement = vector.back();
vector.pop_back();
```

Because although documentation on `std::vector` says that `pop_back()` invalidates iterators and references to the last element/end, we are not using a reference or iterator here. We are copying the value to `current_node` and so deleting from the vector has no impact on `current_node`?

Reply:
Correct.

Reply:
Thank you!